myFileNavLite/Controllers/DownloadController.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  DownloadController.swift
+//  myFileNavLight
+//
+//  Created by Елизавета on 25.05.2020.
+//  Copyright © 2020 Elizaveta. All rights reserved.
+//
+
+import UIKit
+
+class DownloadController: UIViewController {
+    
+    var manager: EventManager? = nil
+    private var docsBuffer: [Document] = []
+    private var links: [(URL?,Bool)] = []
+    
+    @IBOutlet weak var collectionView: UICollectionView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        getLinks()
+        collectionView.register(UINib(nibName: DownloadCell.reuseId, bundle: nil), forCellWithReuseIdentifier: DownloadCell.reuseId)
+    }
+    
+    func getLinks() {
+        let rawLink = Bundle.main.decode([JSONData].self, from: "data.json")
+        for i in 0..<rawLink.count {
+            let str = rawLink[i].url
+            links.append((URL.init(string:str)!,false))
+        }
+    }
+    
+    @IBAction func downloadAll(_ sender: Any) {
+        var isfileDownload = true
+        DispatchQueue.global().async {
+            let dg = DispatchGroup()
+            dg.enter()
+            DispatchQueue.concurrentPerform(iterations: self.links.count) { (i) in
+                if(!self.links[i].1) {
+                    isfileDownload = false
+                    self.downloadDoc(indexPath: IndexPath(item: i, section: 0),downloadAll: true)
+                }
+            }
+            dg.leave()
```

Comment:
This doesn't look like it works correctly :)

Reply:
I'm not sure too, but it works!